src/build_options.rs

```diff
@@ -176,9 +177,9 @@ impl BuildOptions {
             // Only build a source distribution
             Some(ref interpreter) if interpreter.is_empty() => vec![],
             // User given list of interpreters
-            Some(interpreter) => find_interpreter(&bridge, &interpreter, &target)?,
+            Some(interpreter) => find_interpreter(&bridge, &interpreter, &target, None)?,
```

Comment:
Should we also filter the user given list of interpreters based if `requires-python` specified?

Reply:
No, in that case I'd trust the users list more than the PEP 440 heuristic